sklearn/manifold/t_sne.py

```diff
@@ -423,9 +423,11 @@ def trustworthiness(X, X_embedded, n_neighbors=5, precomputed=False):
         dist_X = X
     else:
         dist_X = pairwise_distances(X, squared=True)
-    dist_X_embedded = pairwise_distances(X_embedded, squared=True)
     ind_X = np.argsort(dist_X, axis=1)
-    ind_X_embedded = np.argsort(dist_X_embedded, axis=1)[:, 1:n_neighbors + 1]
+    neigh_embedded = NearestNeighbors(n_neighbors).fit(X_embedded)
+    ind_X_embedded = neigh_embedded.kneighbors(
+        X_embedded, n_neighbors + 1, return_distance=False
+    )[:, 1:]
```

Comment:
Taking `[:, 1:]` doesn't work when there are duplicate points (and hence duplicated zero-distances). This is why we support `kneighbors(1)`. The same issue actually is a bug in the existing code.

Reply:
(fixed a typo, in case you're reading this in your email)

Reply:
> The same issue actually is a bug in the existing code.

I should really read all comments in an issue before doing anything, it took me a little while to figure out why the implementation in this PR had small differences compared to master :-(